app-libs/stf/src/test_genesis.rs

```diff
@@ -90,14 +90,14 @@ fn append_funded_alice_evm_account(endowees: &mut Vec<(AccountId32, Balance, Bal
 #[cfg(not(feature = "evm"))]
 fn append_funded_alice_evm_account(_: &mut Vec<(AccountId32, Balance, Balance)>) {}
 
-fn set_sudo_account(state: &mut SgxExternalities, account_encoded: &[u8]) {
+fn set_sudo_account<State: SgxExternalitiesTrait>(state: &mut State, account_encoded: &[u8]) {
 	state.execute_with(|| {
 		sp_io::storage::set(&storage_value_key("Sudo", "Key"), account_encoded);
 	})
 }
 
-pub fn endow(
-	state: &mut SgxExternalities,
+pub fn endow<State: SgxExternalitiesTrait>(
+	state: &mut State,
```

Comment:
Just as a small input: In these cases, I found the `impl` syntax to be shorter and clearer:
```
fn endow(state: &mut impl SgxExternalitiesTrait, ... ) 
```